### PR TITLE
Naive validation for BTC address

### LINKED ISSUE #1170: Client allows invalid Bitcoin address to be entered for payment
When the seller puts in their Bitcoin address and marks the product as shipped, there is no check to ensure it is a legitimate Bitcoin address. I put "test" in the field and it didn't stop me. We'll need a check there to prevent user error.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/7985489-client-allows-invalid-bitcoin-address-to-be-entered-for-payment?utm_campaign=plugin&utm_content=tracker%2F1516753&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F1516753&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by hoffmabc**
This should eventually be replaced by a more formal validation of BTC address in JS or Python API method. For now this will eliminate garbage values and only allow proper length and characters.

Fixes #1170

### CODE REVIEW

**Comment by gubatron** [comment]
perhaps just add that the address must start with either '1' or '3'

**Comment by STRML** [comment]
The [bitcoin-address](https://www.npmjs.com/package/bitcoin-address) module has been working for us in production at BitMEX.

I think if we're going to validate we should go whole-hog; there is error-checking built into addresses that is quite valuable to have.

**Comment by hoffmabc** [comment]
Ok I can rework this one, I just didn't want to prioritize this one that much. It's fine though, unless someone else wants to take this one on.

**Comment by gubatron** [comment]
what an honor to have @STRML here, your personal website has been making the rounds.

to use such module I suppose we'd need to have nodejs, is OB using nodejs @hoffmabc 
(if we don't use nodejs for clientside, I see several `require` calls in bitcoin-address' code, perhaps there's a way to include all those dependencies in a single .js and then have proper validation)

**Comment by STRML** [comment]
Yes they can be packaged via browserify or webpack for the Web, or we can
extract them ourselves. I'll take a look

**Comment by STRML** [comment]
In future version of the UI we'll package with webpack and we'll be able to reduce module size more significantly and include these modules without issue.

For now, you can use [this browser-packaged script](https://gist.github.com/STRML/d534984b62b7f6c5faa0)

**Comment by hoffmabc** [comment]
Nice. Then I can just use this. I'll modify the PR.

**Comment by hoffmabc** [comment]
OK @STRML I incorporated that bitcoin address validation as a directive. Works well enough. Please review!

File: html/directives/validateBitcoinAddress.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
+angular.module('app').directive("validateBitcoinAddress", function() {
+    return {
+        require: 'ngModel',
+        link: function(scope, ele, attrs, ctrl) {
+
+            ctrl.$parsers.unshift(function(value) {
+
+                // test and set the validity after update.
+                var valid = window.bitcoinAddress.validate(value);
+                ctrl.$setValidity('validateBitcoinAddress', valid);
```
Comment (STRML) [inline html/directives/validateBitcoinAddress.js]: I think it's crazy that you have to do this twice (a failure in Angular in my opinion). Only thing I'd change is to do the validation in the same way on lines 9/10 and line 19.

**Comment by STRML** [comment]
Looks good. See the comment above (no big deal). Did some searching for a more unified way to validate values on controllers in angular but it appears that this is really how you should do it.

**Comment by hoffmabc** [comment]
Ok @STRML made the updates and stripped some useless comments as well. Once it passes if you could merge that'd be good. Thanks!

**Comment by STRML** [comment]
:shipit:

**Pull request merged**